### PR TITLE
[Scheduler] Get current time from performance.now in non-DOM environments

### PR DESCRIPTION
**Pull request by emilisb**
## Summary

Scheduler makes an assumption that `performance.now` does not exist in a non-DOM environment. React Native is such environment, however, latest RN versions do have support for `performance.now` (https://github.com/facebook/react-native/commit/232517a5740f5b82cfe8779b3832e9a7a47a8d3d).

This PR changes `getCurrentTime` implementation to use `performance.now` if it is defined, and fallback to `Date.now` if the function is missing.

This change is required for React Profiler API to work properly in React Native. Without this change, the Profiler in RN uses `Date` which is not so precise.

## Test Plan

- Tests are passing
- Code lints
- I built `scheduler` and tested the change in a React Native project. Profiler API now takes time from `performance.now` and the precision is greatly improved.

Screenshot 2020-08-05 at 14 42 40

### CODE REVIEW

File: packages/scheduler/src/forks/SchedulerHostConfig.default.js [hunk 1]
```diff
@@ -16,10 +16,24 @@ export let requestPaint;
 export let getCurrentTime;
 export let forceFrameRate;
 
+const isWindowUndefined = typeof window === 'undefined';
+
+const performance = isWindowUndefined ? global.performance : window.performance;
```
Comment (gaearon) [inline packages/scheduler/src/forks/SchedulerHostConfig.default.js:21]: Why is this needed? Why not just use `performance` as a global in the first place?

Comment (emilisb) [reply packages/scheduler/src/forks/SchedulerHostConfig.default.js:21]: This is based on a comment in the previous code.

```
// Capture local references to native APIs, in case a polyfill overrides them.
```

Comment (gaearon) [reply packages/scheduler/src/forks/SchedulerHostConfig.default.js:21]: I wonder if all environments that don't have `window` are guaranteed to have `global`, or not. We don't already use `global` in this file yet.

Comment (gaearon) [reply packages/scheduler/src/forks/SchedulerHostConfig.default.js:21]: For example what if this code is loaded in a worker. Would it crash?

Comment (gaearon) [reply packages/scheduler/src/forks/SchedulerHostConfig.default.js:21]: I would feel better about refactoring to read globals directly from the global scope (without a qualifier) and aliasing them. Kind of like it was before https://github.com/facebook/react/commit/3f2cafe8bef0922525b445f7bd2e23ea75c817cf.

Comment (emilisb) [reply packages/scheduler/src/forks/SchedulerHostConfig.default.js]: Good thinking. I refactored the code. I had to refactor the browser test too, `window` is not considered as global under test environment.

**Comment by gaearon** [comment]
Makes sense. Thanks.
